protocol.md

```diff
@@ -159,33 +174,41 @@ Prerequisite: the *user*'s *device* is authenticated against the *Tanker server*
-11. Each *group member*s' *device* retrieves the *block*, verifies it and decrypts the [Group Encryption Key Pair] and [Group Signature Key Pair] using their private [User Encryption Key Pair]
+1. *Tanker Core* fetches all future *group member*s' `device_creation` *block*s from the *Trustchain*
+1. *Tanker Core* verifies them and extracts their public [User Encryption Key Pair]
+1. *Tanker Core* generates the [Group Encryption Key Pair] and the [Group Signature Key Pair]
+1. *Tanker Core* encrypts the private [Group Signature Key Pair] with the public [Group Encryption Key Pair]
+1. *Tanker Core* encrypts the private [Group Encryption Key Pair] with each future *group member*'s public [User Encryption Key Pair]
+1. Using the private [Group Signature Key Pair], *Tanker Core* signs the public and encrypted private [Group Encryption Key Pair] and [Group Signature Key Pair]
+1. *Tanker Core* creates a `user_group_creation` *block* with all of the above and pushes it to the *Trustchain*
+1. The *Tanker server* validates the *block* and sends the update to all *group member*s' *device*s
+1. Each *group member*s' *device* retrieves the *block*, verifies it and decrypts the [Group Encryption Key Pair] and [Group Signature Key Pair] using their private [User Encryption Key Pair]
 
 ### Sharing with user groups
 
 Prerequisite: the *user*'s *device* is authenticated against the *Tanker server*, and some *data* has been encrypted.
 
-Sharing with a *user group* is pretty much the same as sharing with a *user* but requires using a [GID](#group-id) returned by `tanker.createGroup`.
+Sharing with a *user group* is pretty much the same as sharing with a *user* but requires using a [GID](#group-id).
 The steps are as follows:
 
-1. The *application* calls `tanker.share` with the [GID](#group-id) to share with
-2. *Tanker Core* fetches the recipient *group*'s public [Group Encryption Key Pair] from the *Trustchain*, if not already present in the [Local Encrypted Storage]
-3. *Tanker Core* encrypts the [Resource Encryption Key] with the public [Group Encryption Key Pair]. The result is the [Shared Encrypted Key] for this particular *user group* and *resource*
-4. *Tanker Core* creates a *block* containing the [Shared Encrypted Key] and the recipient public [Group Encryption Key Pair]
-5. *Tanker Core* pushes the *block* to the *Trustchain*
-6. The *Tanker server* validates the *block* and notifies all the recipients' *device*s
-8. The recipients' *device*s retrieve the *block*, verify it and decrypt the [Shared Encrypted Key] using the private [Group Encryption Key Pair], obtaining the [Resource Encryption Key]
-7. The *application* calls `tanker.decrypt` on one of the recipient group's *device*s
-10. *Tanker Core* decrypts the data using the [Resource Encryption Key]
+1. The *application* fetches the [GID](#group-id) to share with
+1. *Tanker Core* look for the [Group Encryption Key Pair] associated with [GID](#group-id) in the [Local Encrypted Storage]
```

Comment:
This is the first time we talk about looking things up in the local cache before doing stuff, but we do it for users and resource keys too. We should say that everywhere, or nowhere, but not just here.

also looks*

Reply:
Question is, does it have an impact on the protocol security, like the on the demand ?